src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -1161,27 +1158,54 @@ private bool ConsumeStringAndValidateMultiSegment(ReadOnlySpan<byte> data, int i
             return true;
         }
 
+        private readonly struct PartialStateForRollback
```

Comment:
Mainly using this struct to pass around the relevant states between the methods (to roll back to in case of error OR if we run out of data). I currently pass it by ref, but I could pass it by in since all the fields are readonly. If someone has a suggestion for a different/cleaner approach, let me know.

Reply:
Ideally the nested type would be at the top or bottom of this file, or in a partial file.  Anywhere but the middle.

Reply:
I do feel like passing this struct as `in` would be better than `ref`.  You won't hit bad copies and it will avoid the visual scan that suggests that some of the helper routines might be replacing it.